src/coreclr/vm/threadsuspend.cpp

```diff
@@ -5862,9 +5879,15 @@ void HandleSuspensionForInterruptedThread(CONTEXT *interruptedContext)
 
         pThread->PulseGCMode();
 
-        frame.Pop(pThread);
+        INSTALL_MANAGED_EXCEPTION_DISPATCHER;
+        INSTALL_UNWIND_AND_CONTINUE_HANDLER;
 
         pThread->HandleThreadAbort();
+
+        UNINSTALL_UNWIND_AND_CONTINUE_HANDLER;
+        UNINSTALL_MANAGED_EXCEPTION_DISPATCHER;
+
+        frame.Pop(pThread);
```

Comment:
Should we pop the frame right after `PulseGCMode()` ? 
Is the frame used by the thread abort throw?

Reply:
Yes, it is used by the thread abort throw, that's why it is at this place.